packages/netlify-cms-backend-gitlab/src/API.js

```diff
@@ -185,7 +185,11 @@ export default class API {
   toBase64 = str => Promise.resolve(Base64.encode(str));
   fromBase64 = str => Base64.decode(str);
   uploadAndCommit = async (item, { commitMessage, updateFile = false, branch = this.branch, author = this.commitAuthor }) => {
-    const content = get(item, 'toBase64', partial(this.toBase64, item.raw))();
+    let content = result(item, 'toBase64', partial(this.toBase64, item.raw));
+    if (content instanceof Promise) {
+      content = await content;
+    }
+
```

Comment:
We can remove the `if` check here and just do `await result(...` directly.


> If the value of the *expression* following the `await` operator is not a `Promise`, it's converted to a resolved Promise.
>
> <sup>~https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await</sup>

Reply:
Good to know. Added a commit to change this.